### PR TITLE
Correctly handle active menu states in the new menu markup

### PR DESCRIPTION
**Pull request by aschempp**
Since Contao 4.4, the backend menu requires extensions to always render submenu items. I have fixed this in the core by always passing `$blnShowAll=true` to the hooks and then handle collapsing in the navigation method. This also correctly marks active items if they follow the default naming, which were previously ignored in extensions.

### CODE REVIEW

File: src/Resources/contao/classes/BackendUser.php [hunk 1]
```diff
@@ -484,15 +484,7 @@ public function navigation($blnShowAll=false)
...
-				$arrModules[$strGroupName]['isClosed'] = false;
-
-				// Do not show the modules if the group is closed
-				if (!$blnShowAll && isset($session['backend_modules'][$strGroupName]) && $session['backend_modules'][$strGroupName] < 1)
-				{
-					$arrModules[$strGroupName]['class'] = ' node-collapsed';
-					$arrModules[$strGroupName]['title'] = \StringUtil::specialchars($GLOBALS['TL_LANG']['MSC']['expandNode']);
-					$arrModules[$strGroupName]['isClosed'] = true;
-				}
+				$arrModules[$strGroupName]['icon'] = 'modPlus.gif';
```
Comment (leofeyer) [inline src/Resources/contao/classes/BackendUser.php]: Why did you add this? We are no longer using the GIF icons.

File: src/Resources/contao/classes/BackendUser.php [hunk 2]
```diff
@@ -523,7 +508,33 @@ public function navigation($blnShowAll=false)
 			foreach ($GLOBALS['TL_HOOKS']['getUserNavigation'] as $callback)
 			{
 				$this->import($callback[0]);
-				$arrModules = $this->{$callback[0]}->{$callback[1]}($arrModules, $blnShowAll);
+				$arrModules = $this->{$callback[0]}->{$callback[1]}($arrModules, true);
+			}
+		}
+
+		foreach ($arrModules as $strGroupName => $arrGroupModules)
```
Comment (leofeyer) [inline src/Resources/contao/classes/BackendUser.php]: Why did you move the code?

File: src/Resources/contao/classes/BackendUser.php (hunk 1 above)
Comment (aschempp) [inline src/Resources/contao/classes/BackendUser.php]: Because that was the/a way to detect if a menu is active in previous versions. [EasyThemes](https://github.com/terminal42/contao-easy_themes) relies on it to know if a menu is open.

File: src/Resources/contao/classes/BackendUser.php (hunk 2 above)
Comment (aschempp) [inline src/Resources/contao/classes/BackendUser.php]: Because that code needs to be applied for new menus added by the hook as well?

File: src/Resources/contao/classes/BackendUser.php (hunk 1 above)
Comment (leofeyer) [inline src/Resources/contao/classes/BackendUser.php]: But we have removed this in version 4.2 already (see f49802b0facc6f7e93411936789b025c7aa835cc) and I don't think that we should re-add it.

File: src/Resources/contao/classes/BackendUser.php [hunk 3]
```diff
@@ -523,7 +508,33 @@ public function navigation($blnShowAll=false)
...
+		{
+			$arrModules[$strGroupName]['isClosed'] = false;
+
+			// Do not show the modules if the group is closed
+			if (!$blnShowAll && isset($session['backend_modules'][$strGroupName]) && $session['backend_modules'][$strGroupName] < 1)
+			{
+				$arrModules[$strGroupName]['class'] = str_replace('node-expanded', '', $arrModules[$strGroupName]['class']) . ' node-collapsed';
```
Comment (leofeyer) [inline src/Resources/contao/classes/BackendUser.php]: Why not `str_replace('node-expanded', 'node-collapsed', $arrModules[$strGroupName]['class'])`?

Comment (aschempp) [inline src/Resources/contao/classes/BackendUser.php]: Because `node-expanded` does not necessarily exist.

**Comment by leofeyer** [comment]
Thank you @aschempp.
